packages/core/src/jobs/job-definitions/documents/runDocumentAtCommit/generateToolResponseMessages.ts

```diff
@@ -0,0 +1,85 @@
+import { ToolCall, Message } from '@latitude-data/compiler'
+import {
+  buildResponseMessage,
+  Commit,
+  DocumentVersion,
+  LogSources,
+  Workspace,
+} from '../../../../browser'
+import { Result } from '../../../../lib'
+import { AutogenerateToolResponseCopilotData } from './getCopilotData'
+import { runDocumentAtCommit } from '../../../../services/commits/runDocumentAtCommit'
+import { ToolCallResponse } from '@latitude-data/constants'
+
+function buildIdentifier({
+  workspace,
+  commit,
+  document,
+}: {
+  workspace: Workspace
+  commit: Commit
+  document: DocumentVersion
+}) {
+  return `aigentool-w:${workspace.id}-cmt:${commit.id}-doc:${document.documentUuid}`
+}
+
+export async function generateToolResponseMessages({
+  workspace,
+  commit,
+  document,
+  toolCalls,
+  previousMessages,
+  copilot,
+  source,
+}: {
+  workspace: Workspace
+  commit: Commit
+  document: DocumentVersion
+  toolCalls: ToolCall[]
+  previousMessages: Message[]
+  source: LogSources
+  copilot: AutogenerateToolResponseCopilotData
+}) {
+  const customIdentifier = buildIdentifier({
+    workspace,
+    commit,
+    document,
+  })
+  const result = await runDocumentAtCommit({
+    workspace: copilot.workspace,
+    commit: copilot.commit,
+    document: copilot.document,
+    parameters: { previousMessages, toolCalls },
+    customIdentifier,
+    source,
+  })
+
+  if (result.error) return result
+
+  const responseResult = await result.value.response
+
+  if (responseResult.error) {
+    return Result.error(responseResult.error)
+  }
+  const response = responseResult.value
+  if (response.streamType === 'text') {
+    return Result.error(
+      new Error(
+        'Invalid response. AI responed with text while generating a mock tool response',
+      ),
+    )
+  }
+
+  const toolCallResponses = response.object as ToolCallResponse[]
```

Comment:
Validate that the returned object is like an array of `ToolCallResponse.` AI could hallucinate